src/unix/bsd/freebsdlike/freebsd/mod.rs

```diff
@@ -127,6 +153,679 @@ s! {
         pub shm_dtime: ::time_t,
         pub shm_ctime: ::time_t,
     }
+
+    // SCTP support
+    // netinet/sctp.h
+
+    #[repr(packed)]
+    pub struct sctphdr {
+        /// source port
+        pub src_port: ::uint16_t,
+        /// destination port
+        pub dest_port: ::uint16_t,
+        /// verification tag of packet
+        pub v_tag: ::uint32_t,
+        /// Adler32 checksum
+        pub checksum: ::uint32_t,
+    }
+
+    /// SCTP chunk header
+    #[repr(packed)]
+    pub struct sctp_chunkhdr {
+        /// chunk type
+        pub chunk_type: u8,
+        /// chunk flags
+        pub chunk_flags: u8,
+        /// chunk length
+        pub chunk_length: ::uint16_t,
+    }
+
+    /// SCTP chunk parameters
+    #[repr(packed)]
+    pub struct sctp_paramhdr {
+        /// parameter type
+        pub param_type: ::uint16_t,
+        /// parameter length
+        pub param_length: ::uint16_t,
+    }
+
+    #[repr(packed)]
+    pub struct sctp_gen_error_cause {
+        pub code: ::uint16_t,
+        pub length: ::uint16_t,
+        pub info: [::uint8_t]
```

Comment:
Oh this I don't think is FFI-safe or possible for us to define, for something like this historically I've just defined a 0-length array here instead of a DST array

Reply:
Does fixing this fix the all the remaining errors?

Reply:
I can check to see if using a zero-length array instead fixes the errors, but why would using the flexible / dynamically-sized arrays be FFI-unsafe? If the definition of the ``struct`` in the Rust code matches what's in the C header, isn't that all that's needed for FFI-compatibility?

It seems much more unsafe to use a zero-length array -- consumers of the type will need to use that array like a raw pointer. If the struct is moved or copied somewhere else in memory, the array data won't be copied along with it, and consumers of the struct won't have any real way to tell whether they're looking at the "original" struct with valid data in the array or a copy that's been moved and now has invalid/garbage data where it's expecting the array to be.

Reply:
Sorry I don't have a lot of time to go in depth, but the errors here basically show how a dynamically sized array is a no-go. If it worked that'd be great and for sure better than a zero-length array, but I don't believe it works.